### PR TITLE
Replace non valid values with NaN instead of "" in csv export

### PR DESCRIPTION
**Pull request by pichim**
Edited:
null and other empty non-numeric values are new replaced by a NaN in the *.csv export. On master these values are replaced by "" which leads to problems with some offline data analysis tools like octave.

I would be glad to get some recommendation to the implementation of the change since i don't really understand javascript in detail.

I tested this with several new / newer *.bbl files and it worked as intended.

### CODE REVIEW

**Comment by nerdCopter** [comment]
instead of new code, does modifying line 12 work the same way? (i did not test)
https://github.com/betaflight/blackbox-log-viewer/blob/111b099b80bf10db236d14fa61caee52f6f6f413/js/webworkers/csv-export-worker.js#L12

**Comment by pichim** [comment]
nope :-)

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Review by haslinghuis — APPROVED** [review]

**Comment by ledvinap** [comment]
Emitting "NaN" (or Octave equivalent) won't help? Replacing NaN with 0 in octave is trivial and it won't mask gaps in data (if empty value can be be caused by data gap)

**Comment by pichim** [comment]
@ledvinap good point will test this asap

**Comment by pichim** [comment]
@ledvinap Works with NaN in octave and matlab, thx!

**Comment by haslinghuis** [comment]
Not sure about the code smell here: https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=660&id=betaflight_blackbox-log-viewer&open=AYsQrWe9AlPSDAkREJ6t&tab=why

**Comment by pichim** [comment]
Smelly... :-)

**Comment by pichim** [comment]
No more smells... checked with various *.bbl files.

File: js/webworkers/csv-export-worker.js
Comment (ledvinap) [inline js/webworkers/csv-export-worker.js:39]: You probably want to emit NaN only for falsy values - if string is passed, it shall not be discarded 

```suggestion
                (_.isNumber(value) || _.value)
                ? value
                : "NaN")
```

**Comment by pichim** [comment]
@ledvinap thx, i have absolutely no idea what im doing in java script :-). quickly tested with various *.bbl files and its as expected.

**Review by nerdCopter — APPROVED** [review]

**Comment by nerdCopter** [comment]
tested a handful of BBL, exports to CSV worked including some `NaN`.

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**
